### PR TITLE
Enable Qt5 by default.

### PR DESCRIPTION
**Pull request by rryan**

### CODE REVIEW

**Comment by daschuer** [comment]
What Implications we have choosing the QT version? 

14.4 Trusty  April 2019  Qt 5.2
16.04 Xenial April 2021 Qt 5.5
17.10 Artful Juli 2018 Qt 5.9
18.04 Bionic April 2023 Qt 5.9 

My dev system still runs at Trutsy mainly to test this setup. 
Are there strong reasons to require Xenial and Qt 5.5? 

I think supporting two LTS is enough, but we should not artificially break still supported versions.

**Comment by rryan** [comment]
Mixxx doesn't build at head on qt 5.2. Probably not for a good reason, but
preserving compatibility with old versions will in general slow us down.
There are almost 2 LTS releases newer than Trust, and 2.0/2.1 are always
available to those users.

File: build/appveyor/build_mixxx.bat [hunk 1]
```diff
@@ -140,7 +140,7 @@ set CXXFLAGS=/MP /FS /EHsc /Zc:threadSafeInit-
 set CFLAGS=/MP /FS /EHsc /Zc:threadSafeInit-
 
 set PATH=%BIN_DIR%;%PATH%
-scons.py mixxx makerelease toolchain=msvs winlib=%WINLIB_DIR% build=%BUILD_TYPE% staticlibs=1 staticqt=1 debug_assertions_fatal=1 verbose=0 machine=%MACHINE_TYPE% qtdir=%QTDIR% hss1394=1 mediafoundation=1 opus=1 localecompare=1 optimize=fastbuild virtualize=0 test=1 qt_sqlite_plugin=0 mssdk_dir="%MSSDK_DIR%" build_number_in_title_bar=0 bundle_pdbs=1
+scons.py mixxx makerelease qt5=0 toolchain=msvs winlib=%WINLIB_DIR% build=%BUILD_TYPE% staticlibs=1 staticqt=1 debug_assertions_fatal=1 verbose=0 machine=%MACHINE_TYPE% qtdir=%QTDIR% hss1394=1 mediafoundation=1 opus=1 localecompare=1 optimize=fastbuild virtualize=0 test=1 qt_sqlite_plugin=0 mssdk_dir="%MSSDK_DIR%" build_number_in_title_bar=0 bundle_pdbs=1
```
Comment (Pegasus-RPG) [inline build/appveyor/build_mixxx.bat]: Probably should comment here why you're setting qt5=0 when this PR is about enabling Qt5. ;)

Comment (rryan) [reply build/appveyor/build_mixxx.bat]: Done

**Comment by rryan** [comment]
> I'm baffled as to why the Linux build fails while this builds fine on macOS. The Qt5 incompatibilities (QAtomicInt/QAtomicPointer) aren't reverted in 5.10.1 AFAIK.

It builds fine for me on my Xenial machine.

**Comment by rryan** [comment]
`fatal error LNK1318: Unexpected PDB error; OK (0)`

Ok MSVC.

**Comment by rryan** [comment]
I think the AppVeyor build is failing because it's running out of RAM (even though it's in fastbuild mode).

**Comment by Pegasus-RPG** [comment]
That might be it: https://stackoverflow.com/questions/4256524/lnk1318-unexpected-pdb-error-ok-0#5105245

**Comment by rryan** [comment]
> That might be it: https://stackoverflow.com/questions/4256524/lnk1318-unexpected-pdb-error-ok-0#5105245

AppVeyor is always building from a totally clean environment though. I still think it's just running out of RAM and MSVC is giving us a garbage error message. You can find some instances of people hitting this error saying they were out of RAM.

**Comment by rryan** [comment]
Disabling debug information seems to make it work, which is non-ideal.

x64 succeeded: https://ci.appveyor.com/api/buildjobs/vxf1l1q2794dkcn3/artifacts/mixxx-2.2.0-alpha-pre-master-PR1632-git6491-release-x64.exe

**Comment by Pegasus-RPG** [comment]
So it's likely a memory issue then. What is the limit on our Appveyor instance? Can we make any changes to the appveyor.yml file to reduce memory use? Maybe something to disable mspdbsrv.exe as mentioned in that Stack Overflow article? (Unless that's needed to generate the PDBs.)

**Comment by Be-ing** [comment]
Any ideas why this takes more memory than Qt4?

File: build/debian/control [hunk 2]
```diff
@@ -6,7 +6,7 @@ Build-Depends: debhelper (>= 9),
                pkg-config,
                docbook-to-man,
                libglu1-mesa-dev,
-               libqt4-opengl-dev | libqt4-dev (>= 4.6.0),
+               libqt5opengl5-dev | qt5-default (>= 5.2.0),
```
Comment (Be-ing) [inline build/debian/control]: Have you tested that Mixxx actually works with Qt 5.2?

Comment (rryan) [reply build/debian/control]: Travis uses Trusty which is on 5.2, so it builds at least.

Comment (daschuer) [reply build/debian/control]: I have still some build issues on Trusty with Qt 5.2 a PR is on the way.

**Comment by rryan** [comment]
> Any ideas why this takes more memory than Qt4?

I'm not sure, a couple hypotheses:
* All development moved to Qt5 long ago, and Qt5 is probably larger in terms of binary size and number of debug symbols because of it.
* Qt5 added more principled PDB-in-release-mode options, whereas with Qt4 we were kind of hacking them by copying intermediate PDBs out of the build directories. It's possible we were missing some PDBs in Qt4, or they didn't contain all the debug info that Qt5 does.

**Comment by Be-ing** [comment]
Ready for merge?

**Comment by rryan** [comment]
Note that #1678 fixes issues that I think are already fixed by this PR, since Travis builds with Qt 5.2 on Trusty and is green.

**Comment by rryan** [comment]
> Ready for merge?

Yup! Just waiting on a review.

File: build/depends.py [hunk 3]
```diff
@@ -409,11 +446,33 @@ def configure(self, build, conf):
...
+                        os.path.join(build.env['QTDIR'],'plugins/platforms')])
+                    build.env.Append(LIBS = 'qwindows')
+
+                # styles (new in Qt5 for Windows)
+                if qt5:
+                    build.env.Append(LIBPATH=[
+                        os.path.join(build.env['QTDIR'],'plugins/styles')])
+                    build.env.Append(LIBS = 'qwindowsvistastyle')
+
+                # sqldrivers (new in Qt5? or did we just start enabling them)
```
Comment (Pegasus-RPG) [inline build/depends.py]: They existed in Qt4 but we disabled them because we link directly with SQLite so it includes our build flags/enhancements.

**Comment by Pegasus-RPG** [comment]
What are all the changes in playermanager.cpp? They seem unrelated, or that they belong in another PR at first glance.

**Comment by rryan** [comment]
> What are all the changes in playermanager.cpp? They seem unrelated, or that they belong in another PR at first glance.

Yea, they're all build fixes for Qt 5 -- we had a bunch of Qt 4-specific code slip in at some point.

**Comment by Pegasus-RPG** [comment]
Do we still want to build our own SQLite? If so, we shouldn't use the Qt5 plugin.

**Comment by rryan** [comment]
> Do we still want to build our own SQLite? If so, we shouldn't use the Qt5 driver.

We don't use SQLite directly in Mixxx other than to setup custom collations -- we're always using it through QtSQL, so we always need the SQLite QtSql driver. The SQLite QtSql driver plugin is linked to the system sqlite library (read: Mixxx's version of sqlite), so I *think* we're all good here.

**Comment by rryan** [comment]
The SQLite symbols are all undefined in this lib, so I think we're all good:
```
$ nm 2.2-j00009-x86-release-fastbuild-static-ed0283e-minimal/Qt-5.10.1/plugins/sqldrivers/qsqlite.lib
         U _memcpy
         U _sqlite3_bind_blob
         U _sqlite3_bind_double
         U _sqlite3_bind_int
         U _sqlite3_bind_int64
         U _sqlite3_bind_null
         U _sqlite3_bind_parameter_count
         U _sqlite3_bind_parameter_name
         U _sqlite3_bind_text16
         U _sqlite3_busy_timeout
         U _sqlite3_changes
         U _sqlite3_close
         U _sqlite3_column_blob
         U _sqlite3_column_bytes
         U _sqlite3_column_bytes16
         U _sqlite3_column_count
         U _sqlite3_column_decltype16
         U _sqlite3_column_double
         U _sqlite3_column_int
         U _sqlite3_column_int64
         U _sqlite3_column_name16
         U _sqlite3_column_table_name16
         U _sqlite3_column_text16
         U _sqlite3_column_type
         U _sqlite3_create_function_v2
         U _sqlite3_errmsg16
         U _sqlite3_finalize
         U _sqlite3_last_insert_rowid
         U _sqlite3_open_v2
         U _sqlite3_prepare16_v2
         U _sqlite3_reset
         U _sqlite3_result_int
         U _sqlite3_step
         U _sqlite3_update_hook
         U _sqlite3_user_data
         U _sqlite3_value_text
```

File: build/depends.py (hunk 3 above)
Comment (rryan) [reply build/depends.py]: Note, this is a static plugin, not a DLL like the `qt_sqlite_plugin` scons parameter controls. I believe in Qt 4 static builds, the QtSql SQLite driver is built-in to Qt, and in Qt 5 when static built it has been moved to a static sqldriver plugin. Could be wrong...

**Pull request merged**